src/extension.ts

```diff
@@ -35,95 +35,101 @@ if (!global.fetch) {
+		context.subscriptions.push(commandManager);
 
-	// Create auth provider as soon as possible
-	const authProvider = new GitpodAuthenticationProvider(context, logger, telemetryService, notificationService);
-	context.subscriptions.push(authProvider);
+		// Create auth provider as soon as possible
+		const authProvider = new GitpodAuthenticationProvider(context, logger, telemetryService, notificationService);
+		context.subscriptions.push(authProvider);
 
-	const hostService = new HostService(context, notificationService, logger);
-	context.subscriptions.push(hostService);
+		const hostService = new HostService(context, notificationService, logger);
+		context.subscriptions.push(hostService);
 
-	const sessionService = new SessionService(hostService, logger);
-	context.subscriptions.push(sessionService);
+		const sessionService = new SessionService(hostService, logger);
+		context.subscriptions.push(sessionService);
 
-	const experiments = new ExperimentalSettings('gitpod', packageJSON.version, sessionService, context, logger);
-	context.subscriptions.push(experiments);
+		const experiments = new ExperimentalSettings('gitpod', packageJSON.version, sessionService, context, logger);
+		context.subscriptions.push(experiments);
 
-	const settingsSync = new SettingsSync(commandManager, logger, telemetryService, notificationService);
-	context.subscriptions.push(settingsSync);
+		const settingsSync = new SettingsSync(commandManager, logger, telemetryService, notificationService);
+		context.subscriptions.push(settingsSync);
 
-	const remoteConnector = new RemoteConnector(context, sessionService, hostService, experiments, logger, telemetryService, notificationService);
-	context.subscriptions.push(remoteConnector);
+		const remoteConnector = new RemoteConnector(context, sessionService, hostService, experiments, logger, telemetryService, notificationService);
+		context.subscriptions.push(remoteConnector);
 
-	context.subscriptions.push(vscode.window.registerUriHandler({
-		handleUri(uri: vscode.Uri) {
-			// logger.trace('Handling Uri...', uri.toString());
-			if (uri.path === GitpodServer.AUTH_COMPLETE_PATH) {
-				authProvider.handleUri(uri);
-			} else {
-				remoteConnector.handleUri(uri);
+		context.subscriptions.push(vscode.window.registerUriHandler({
+			handleUri(uri: vscode.Uri) {
+				// logger.trace('Handling Uri...', uri.toString());
+				if (uri.path === GitpodServer.AUTH_COMPLETE_PATH) {
+					authProvider.handleUri(uri);
+				} else {
+					remoteConnector.handleUri(uri);
+				}
 			}
+		}));
+
+		context.subscriptions.push(new ReleaseNotes(context, commandManager, logger));
+
+		// Register global commands
+		commandManager.register(new SignInCommand(sessionService));
+		commandManager.register(new ExportLogsCommand(context.logUri, notificationService, telemetryService, logger));
+
+		const remoteConnectionInfo = getGitpodRemoteWindowConnectionInfo(context);
+		telemetryService.sendTelemetryEvent('vscode_desktop_activate', {
```

Comment:
we shoudl mode it to finally block and `success: boolean` property to indicate whether acitaved or not